app/themes/default/components/ui/admin/ProductCategoryList.tsx

```diff
@@ -1,12 +1,209 @@
+
+const ProductCategoryList = ({
+  categories,
+}: {
+  categories: CategoryItem[]
+}) => {
   const { t } = useTranslation()
   const { storeSettings } = useContext(AdminContext)
 
-  return <></>
-}
+  const submit = useSubmit()
+  const fetcher = useFetcher()
+
+  const [editOpen, setEditOpen] = useState(false)
+  const [isCreate, setIsCreate] = useState(false)
+  const [formData, setFormData] = useState<{
+    [key: string]: string | null
+  }>({
+    name: '',
+    slug: '',
+    intent: '',
+    parentId: '',
+  })
+
+  useEffect(() => {
+    if (fetcher.data && !fetcher.data.error) {
+      setEditOpen(false)
+    }
+  }, [fetcher.data])
 
+  return (
+    <>
+      <Button
+        className="bg-green-600 text-black float-right"
+        onClick={() => {
+          setFormData({
+            name: '',
+            slug: '',
+            intent: 'create-category',
+            parentId: '',
+          })
+
+          setEditOpen(true)
+          setIsCreate(true)
+        }}
+      >
+        {t('system.add')}
+      </Button>
+      {categories && categories.length ? (
+        <Table>
+          <TableHeader>
+            <TableRow>
+              <TableHead>{t('system.category')}</TableHead>
+            </TableRow>
+          </TableHeader>
+
+          <TableBody>
+            {categories.map((category) => (
+              <TableRow key={category.id}>
+                <TableCell>{category.id}</TableCell>
+                <TableCell className="font-medium">{category.name}</TableCell>
+                <TableCell>
+                  <Button
+                    type="button"
+                    variant="link"
+                    onClick={() => {
+                      setFormData({ ...category, intent: 'update-category' })
+                      setIsCreate(false)
+                      setEditOpen(true)
+                    }}
+                  >
+                    {t('system.edit')}
+                  </Button>
+                </TableCell>
+              </TableRow>
+            ))}
+          </TableBody>
+        </Table>
+      ) : (
+        <div className="text-center pt-32">{t('system.no_records_found')}</div>
+      )}
+      <Dialog
+        open={editOpen}
+        onOpenChange={() => {
+          setEditOpen(!editOpen)
+        }}
+      >
+        <DialogContent>
+          <DialogHeader className="max-w-screen-sm">
+            <DialogTitle>
+              {isCreate
+                ? t('system.create_new_category')
+                : t('system.edit_category')}
+            </DialogTitle>
+          </DialogHeader>
+          <fetcher.Form
+            onSubmit={() => {
+              submit(formData, { method: 'POST' })
+            }}
+            encType="application/x-www-form-urlencoded"
+            className="space-y-4"
+          >
+            <div>
+              <Input
+                type="hidden"
+                id="category-id"
+                name="category-id"
+                value={''}
```

Comment:
the value shoule be also from formData, otherwise how can you set the **category-id** for editing a category?